### PR TITLE
bazel_4: 4.2.1 -> 4.2.2

### PR DESCRIPTION
**Pull request by timothyklim**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
cc @NixOS/bazel

**Comment by uri-canva** [comment]
I'm confused by the new `bazel_4, bazel_4.passthru.tests` ofborg build, the log looks like it only built `bazel_4.passthru.tests`, and not `bazel_4` separately.

File: pkgs/development/tools/build-managers/bazel/bazel_4/default.nix [hunk 1]
```diff
@@ -214,14 +214,6 @@ stdenv.mkDerivation rec {
...
       bazelSystemBazelRCPath = bazelRC;
     })
-
-    # On macOS Monterey, protoc segfaults.
-    # Issue: https://github.com/bazelbuild/bazel/issues/14216
-    # Fix: https://github.com/bazelbuild/bazel/pull/14275
-    (fetchpatch {
-      url = "https://github.com/bazelbuild/bazel/commit/ae0a6c98d4f94abedbedb2d51c27de5febd7df67.patch";
-      sha256 = "sha256-YcdxqjTMGI86k1wgFqxJqghv0kknAjlFQFpt4VccCTE=";
-    })
```
Comment (uri-canva) [inline pkgs/development/tools/build-managers/bazel/bazel_4/default.nix:224]: Why was this removed? The commit wasn't cherry-picked to `4.2.2`.

**Review by uri-canva — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/build-managers/bazel/bazel_4/default.nix (hunk 1 above)
Comment (timothyklim) [reply pkgs/development/tools/build-managers/bazel/bazel_4/default.nix:224]: I didn't remove it until my build stop on error about patch already applied and then I looked at https://github.com/bazelbuild/bazel/blob/master/CHANGELOG.md#release-422-2021-12-02:
```
 + ae0a6c98d4f94abedbedb2d51c27de5febd7df67:
     Enable user_link_flags_feature for macosx cc_toolchain_config
```
it's already included in this release.

Comment (uri-canva) [reply pkgs/development/tools/build-managers/bazel/bazel_4/default.nix:224]: Oh right, the cherry pick would have a different commit hash, all good!

**Review by uri-canva — APPROVED** [review]
